src/Soil-Core/SoilIndexIterator.class.st

```diff
@@ -105,6 +105,17 @@ SoilIndexIterator >> next [
 	^ self nextAssociation value
 ]
 
+{ #category : #accessing }
+SoilIndexIterator >> next: anInteger [
+	| result |
+	result := OrderedCollection new: anInteger.
+	anInteger timesRepeat: [
```

Comment:
If the remaining items are smaller then the requested anInteger is it intended that the resulting OrderedCollection has the size of anInteger with trailing nils instead of an OrderedCollection that has the size of the remaining items